src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/HeterogeneousPloidyModel.java

```diff
@@ -17,25 +20,12 @@
     private final boolean isHomogeneous;
 
     public HeterogeneousPloidyModel(final SampleList sampleList, final int[] ploidies) {
-        Utils.nonNull(sampleList, "the sample list cannot be null");
-        Utils.nonNull(ploidies, "the ploidies cannot be null");
-        if (sampleList.numberOfSamples() != ploidies.length) {
-            throw new IllegalArgumentException("sample-list and ploidy array length must match");
-        }
-
-        this.ploidies = ploidies.clone();
-
-        int ploidySum = 0;
-        for (int i = 0; i < ploidies.length; i++) {
-            final int p = this.ploidies[i];
-            if (p < 0) {
-                throw new IllegalArgumentException("no ploidy can be less than 0");
-            }
-            ploidySum += p;
-        }
-        this.ploidySum = ploidySum;
-        isHomogeneous = ploidies.length == 0 || ploidies.length * this.ploidies[0] == ploidySum;
-        this.sampleList = sampleList;
+        this.sampleList = Utils.nonNull(sampleList, "the sample list cannot be null");
+        this.ploidies = Utils.nonNull(ploidies, "the ploidies cannot be null").clone();
+        Utils.validateArg(sampleList.numberOfSamples() == ploidies.length, "sample-list and ploidy array length must match");
+        Arrays.stream(ploidies).forEach(p -> {if (p < 0) throw new IllegalArgumentException("no ploidy can be less than 0");});
+        ploidySum = (int) MathUtils.sum(ploidies);
+        isHomogeneous = ploidies.length == 0 || Arrays.stream(ploidies).allMatch(p -> p == ploidies[0]);
```

Comment:
Previous version was `ploidies.length == 0 || ploidies.length * this.ploidies[0] == ploidySum`, which would not necessarily have required all the ploidies to equal the first ploidy, as in the new version. Is this change intended?